src/httpserver.cpp

```diff
@@ -414,6 +417,7 @@ bool InitHTTPServer()
     }
 
     evhttp_set_timeout(http, GetArg("-rpcservertimeout", DEFAULT_HTTP_SERVER_TIMEOUT));
+    evhttp_set_max_headers_size(http, MAX_HEADERS_SIZE);
```

Comment:
Wait, this PR is only further restricting the headers.
Shouldn't the title be then: "http: Restrict maximum size of headers"